editor/code_editor.cpp

```diff
@@ -1129,6 +1129,27 @@ void CodeTextEditor::trim_trailing_whitespace() {
 	}
 }
 
+void CodeTextEditor::trim_final_newlines() {
+	int final_line = text_editor->get_line_count() - 1;
+	int check_line = final_line;
+
+	String line = text_editor->get_line(final_line);
+
+	while (line.is_empty() && check_line > -1) {
+		line = text_editor->get_line(--check_line);
+	}
+
+	if (check_line++ < final_line) {
```

Comment:
```suggestion
	if (check_line < final_line) {
		++check_line;
```
More readable IMO, it doesn't matter outside of the block anyway

Reply:
That was supposed to be an `if(++check_line < final_line)` to avoid triggering removal when there is zero or one empty lines at the end of the file. 

Is the preference increment/decrement outside of operations, or is this and the `line = text_editor->get_line(--check_line)` fine?

Reply:
I'd say generally yes, it's much better for readability

Reply:
Awesome, I'll update this tonight then we are just waiting on the caret/selection PR to be merged.